File: ITSC 3155/itsc-3155/m1/syntax/classes/classes.java
// a simple review to show how classes work in Java
package m1.syntax.classes;

public class classes {

    // class definition
    static class Person {
        // class body
        String name;
        int age;

        // constructor
        public Person(String name, int age) {
            this.name = name;
            this.age = age;
        }

        // method definition
        public void greeting() {
            System.out.println("Hello, my name is " + name);
        }

        public void sayAge() {
            System.out.println("I am " + age + " years old");
        }
    }

    public static void main(String[] args) {
        // creating an instance of the class (an object)
        Person person1 = new Person("George", 20);
        Person person2 = new Person("Georgia", 21);

        // calling the methods
        person1.greeting();
        person1.sayAge();
        person2.greeting();
        person2.sayAge();

    }
}

File: ITSC 3155/itsc-3155/m1/syntax/classes/classes.py
# a simple review to show how classes work in Python

# class definition
class Person:
    # class body (MUST BE INDENTED ONE TAB)

    # class constructor
    def __init__(self, name: str, age: int):
        # instance variables (attributes)
        self.name = name
        self.age = age

    # method definition (must pass self as first parameter)
    def greeting(self):
        print(f"Hello, my name is {self.name}")

    # another method definition
    def sayAge(self):
        print(f"I am {self.age} years old")


def main():
    # creating an instance of the Person class
    p1 = Person("George", 20)
    p2 = Person("Georgia", 21)

    # calling p1's methods
    p1.greeting()
    p1.sayAge()

    # NOTE: we do not need to pass self as a parameter when calling methods, only when defining them

    # calling p2's methods
    p2.greeting()
    p2.sayAge()


if __name__ == "__main__":
    main()
